htdocs/compta/facture/class/facture.class.php

```diff
@@ -1730,14 +1730,17 @@ public function fetch($rowid, $ref = '', $ref_ext = '', $notused = '', $fetch_si
 					$this->error = $this->db->error();
 					return -3;
 				}
+
+				$this->db->free($resql);
+
 				return 1;
 			} else {
-				$this->error = 'Invoice with id='.$rowid.' or ref='.$ref.' or ref_ext='.$ref_ext.' not found';
-				dol_syslog(get_class($this)."::fetch Error ".$this->error, LOG_ERR);
+				dol_syslog(__METHOD__ . ' Invoice with id=' . $rowid . ' or ref=' . $ref . ' or ref_ext=' . $ref_ext . ' not found', LOG_DEBUG);
 				return 0;
 			}
 		} else {
-			$this->error = $this->db->error();
+			$this->error = $this->db->lasterror();
+			dol_syslog(__METHOD__ . ' Error : ' . $this->error, LOG_ERR);
```

Comment:
There is no need to add a dol_syslog here. When a sql error occurs, there is already automaticaly a dol_syslog to output the error.

Reply:
Thanks for your remark, I removed this line.